src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -2078,7 +2078,7 @@ internal static BinaryOperationKind DeriveBinaryOperationKind(UnaryOperationKind
                     return BinaryOperationKind.DynamicSubtract;
 
                 default:
-                    throw ExceptionUtilities.UnexpectedValue(incrementKind);
+                    return BinaryOperationKind.None;
```

Comment:
@JohnHamby I just did a quick search for references to `BinaryOperationKind.None` in Roslyn, it turns out it has been used in several places. For example
```
BasicCodeAnalysis\BoundTree\Expression.vb - (95, 44) : Return BinaryOperationKind.None
```

It seems they all represent some kind of error state and probably should be replaced with the newly added `BinaryOperationKind.Invalid`, what do you think?